rice-datanode/source/data_transfer_server.cc

```diff
@@ -318,7 +324,121 @@ void TransferServer::synchronize(std::function<void(TransferServer&, tcp::socket
 	cv.notify_one();
 }
 
+bool TransferServer::replicate(uint64_t len, std::string ip, std::string xferport, ExtendedBlockProto blockToTarget) {
```

Comment:
@Swyrough we need to increment and decrement xmits. We also need to block if xmits >= MAX_XMITS. Please refer to the synchronize method